Modix.Bot/Behaviors/CommandListeningBehavior.cs

```diff
@@ -64,7 +73,26 @@ public async Task HandleNotificationAsync(MessageReceivedNotification notificati
 
             await AuthorizationService.OnAuthenticatedAsync(author);
 
-            var commandResult = await CommandService.ExecuteAsync(commandContext, argPos, ServiceProvider);
+            IResult commandResult = null;
+            var commandTimer = Stopwatch.StartNew();
+            try
+            {
+                commandResult = await CommandService.ExecuteAsync(commandContext, argPos, ServiceProvider);
+            }
+            finally
+            {
+                commandTimer.Stop();
+                var duration = commandTimer.ElapsedMilliseconds;
+
+                if (!(_stats is null) && (commandResult.IsSuccess || !string.Equals(commandResult.ErrorReason, "UnknownCommand", StringComparison.OrdinalIgnoreCase)))
+                {
+                    var commandInfo = CommandService.Search(commandContext, argPos).Commands.FirstOrDefault();
+                    var name = commandInfo.Command?.Name.ToLowerInvariant();
```

Comment:
I'm not really happy with how this is implemented right now since we're duplicating work that had already been done by the `CommandService` as part of executing the command. I'm definitely open to iterating on this to find a better way in a future PR.

Reply:
What? What work?

Reply:
I imagine that `CommandService.ExecuteAsync` might make this same call to determine which command to execute. I don't know the internals and didn't have time to look.